python/ray/_private/test_utils.py

```diff
@@ -391,6 +391,34 @@ async def async_wait_for_condition(
     raise RuntimeError(message)
 
 
+async def async_wait_for_condition_async_predicate(
+    async_condition_predictor, timeout=10, retry_interval_ms=100, **kwargs: Any
+):
+    """Wait until a condition is met or time out with an exception.
+
+    Args:
+        condition_predictor: A function that predicts the condition.
+        timeout: Maximum timeout in seconds.
+        retry_interval_ms: Retry interval in milliseconds.
+
+    Raises:
+        RuntimeError: If the condition is not met before the timeout expires.
+    """
+    start = time.time()
+    last_ex = None
+    while time.time() - start <= timeout:
+        try:
+            if await async_condition_predictor(**kwargs):
+                return
+        except Exception as ex:
+            last_ex = ex
```

Comment:
wondering if we should default to no retry? since any failure after a successful retries would hide the intermediate failures, which I feel could be alarming.

Reply:
Oh I think I actually meant to change the default retry so that w/o explicitly retry flag, the test will fail on its first exception.